File: C/runnersMeetings.c
runnersMeetings(arr_integer p, arr_integer s) {
    int r = -1;
    int l = p.size;

    for (int i = 0; i < l; i++) {
        for (int j = 0; j < l; j++) {
            int c = 0;
            int d = p.arr[j] - p.arr[i];
            int f = s.arr[i] - s.arr[j];

            if (d * f < 1)
                continue;

            for (int k = 0; k < l; k++)
                p.arr[k] * f + s.arr[k] * d == p.arr[i] * f + s.arr[i] * d ? c++ : 0;

            c > r ? r = c : 0;
        }
    }
    return r;
}

File: Rust/runnersMeetings.rs
fn runnersMeetings(p: Vec<i32>, s: Vec<i32>) -> i32 {
    let mut r = -1;
    let l = p.len();

    for i in 0..l {
        for j in 0..l {
            let mut c = 0;
            let d = p[j] - p[i];
            let f = s[i] - s[j];

            if d * f < 1 {
                continue;
            }

            for k in 0..l {
                if p[k] * f + s[k] * d == p[i] * f + s[i] * d {
                    c += 1;
                }
            }

            if c > r {
                r = c;
            }
        }
    }
    r // Return
}
